src/screens/Vault/SignerAdvanceSettings.tsx

```diff
@@ -552,6 +559,46 @@ function SignerAdvanceSettings({ route }: any) {
         return;
       }
 
+      if (SignersReqVault.includes(signer.type)) {
+        let activeVault = null;
+        allVaults.forEach(async (vault) => {
+          let addressMatched = true;
+          for (let i = 0; i < senderAddresses.length; i++) {
+            const _ = senderAddresses[i].path.split('/');
+            const [chain, index] = _.splice(_.length - 2);
+            // 0 - Receive(External) | 1 - change(external)
+            let generatedAddress;
+            if (chain == '0') {
+              generatedAddress = WalletOperations.getExternalInternalAddressAtIdx(
+                vault,
+                parseInt(index)
+              );
+            } else {
+              generatedAddress = WalletOperations.getExternalInternalAddressAtIdx(
+                vault,
+                parseInt(index),
+                true
+              );
+            }
```

Comment:
could write it like this:
```suggestion
            if (chain != '0' && chain != '1' {
                throw new Error('Derivation uses an invalid path');
            }
            generatedAddress = WalletOperations.getExternalInternalAddressAtIdx(
              vault,
              parseInt(index),
              chain != '0'
            );
```

Reply:
Also better call it isChange not chain since this is not the chain part